models.py

```diff
@@ -276,16 +277,24 @@ def __init__(
 		self.residual = residual
 		self.dict = dict
 		self.bpe_only = bpe_only
+		self.check_time_dim_padded = check_time_dim_padded
 
 	def forward(
 		self, x: typing.Union[shaping.BCT, shaping.BT], xlen: typing.Optional[shaping.B] = None, y: typing.Optional[shaping.BLY] = None, ylen: typing.Optional[shaping.B] = None
 	) -> shaping.BCt:
 
+		if self.check_time_dim_padded:
+			assert x.shape[-1] % (32 / 2) != 0, 'Shape of input signal is not divisible by 16'
+
 		if self.frontend is not None:
 			x = x.squeeze(1)
 			mask = temporal_mask(x, compute_output_lengths(x, xlen)) if xlen is not None else None
 			x = self.frontend(x, mask = mask)
 			# NOTE: squeeze(1) cause onnxruntime warnings like "Force fallback to CPU execution for node: Gather_3 Force fallback to CPU execution for node: Equal_5".
+
+			if self.check_time_dim_padded:
```

Comment:
Why is this happening inside the `if`? it should be checked regardless of how log_mel_features were computed

Reply:
I pace it inside because in both cases it is no guarantee that before and after fronted dim size will be divisible.

Reply:
why? in training we have collate_fn that ensures divisibility. for benchmarking this should be satisfied by careful choice of shapes. if we want to turn off the check, we have an explicit way of doing so.

please move this check outside the if. anyway, it's just a way of finding some padding bugs

Reply:
done